api/config/v1/flags.go

```diff
@@ -101,12 +101,12 @@ func (f *deviceListStrategyFlag) UnmarshalJSON(b []byte) error {
 
 // GFDCommandLineFlags holds the list of command line flags specific to GFD.
 type GFDCommandLineFlags struct {
-	Oneshot           *bool     `json:"oneshot"         yaml:"oneshot"`
-	NoTimestamp       *bool     `json:"noTimestamp"     yaml:"noTimestamp"`
-	SleepInterval     *Duration `json:"sleepInterval"   yaml:"sleepInterval"`
-	OutputFile        *string   `json:"outputFile"      yaml:"outputFile"`
-	MachineTypeFile   *string   `json:"machineTypeFile" yaml:"machineTypeFile"`
-	VfioPciDriverMode *bool     `json:"vfioPciDriverMode" yaml:"vfioPciDriverMode"`
+	Oneshot         *bool     `json:"oneshot"         yaml:"oneshot"`
+	NoTimestamp     *bool     `json:"noTimestamp"     yaml:"noTimestamp"`
+	SleepInterval   *Duration `json:"sleepInterval"   yaml:"sleepInterval"`
+	OutputFile      *string   `json:"outputFile"      yaml:"outputFile"`
+	MachineTypeFile *string   `json:"machineTypeFile" yaml:"machineTypeFile"`
+	Mode            *string   `json:"mode" yaml:"mode"`
```

Comment:
Question: Is this a top-level flag instead? That is to say, would this mode control both GFD and the device plugin?

Reply:
Does running in the TEGRA mode mean anything to device plugin? Also currently we dont support NVML as well. So I feel it makes sense to keep it here.

Reply:
We have similar logic in the device plugin here: https://github.com/NVIDIA/k8s-device-plugin/blob/cf082174cda71b023030df58a6639572aaaea040/internal/plugin/manager/factory.go#L97-L133

This also changes the way resources are disovered based on autodetecting the platform (nvml, and tegra). I would assume that we would do something similar to allow the device plugin to also serve vfio devices?

Reply:
```suggestion
	Mode            *string   `json:"mode"            yaml:"mode"`
```